src/HotChocolate/Data/src/Data/Projections/Convention/IProjectionFieldInterceptor`1.cs

```diff
@@ -0,0 +1,17 @@
+using HotChocolate.Execution.Processing;
+
+namespace HotChocolate.Data.Projections
+{
+    public interface IProjectionFieldInterceptor<in TContext>
+        : IProjectionFieldInterceptor
+        where TContext : IProjectionVisitorContext
+    {
+        void BeforeProjection(
```

Comment:
@michaelstaib  i am not sure about these methods. 
BeforeProjection is called before TryEnter and TryLeave of the field handler but after OnBeforeEnter. AfterProjection is called after Enter and Leave but before OnAfterLeave

In case of a list it could be that the context is already cleaned up in AfterProjection. This leads to a few things that are not nice. Like it is not possible to add `Take(1)` after `Select(x=> new Foo { Bar = x.Bar)`. It can only be attached before it because in AfterProjection the Scope is already popped and the members assigment is pushed to the context
I am not sure if this is a design issue. What do you think?